apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerException.cs

```diff
@@ -101,7 +122,20 @@ internal SpannerException(ErrorCode code, RpcException innerException)
             : base(GetMessageFromErrorCode(code), innerException)
         {
             Logger.DefaultLogger.LogPerformanceCounter("SpannerException.Count", x => x + 1);
+
+            SessionExpired = innerException.IsSessionExpiredError();
             ErrorCode = innerException.IsSessionExpiredError() ? ErrorCode.Aborted : code;
+
+            Metadata.Entry retryInfoEntry = innerException.Trailers.FirstOrDefault(
+                entry => s_retryInfoMetadataKey.Equals(entry.Key, StringComparison.InvariantCultureIgnoreCase));
+            if (retryInfoEntry != null)
+            {
+                RetryInfo retryInfo = RetryInfo.Parser.ParseFrom(retryInfoEntry.ValueBytes);
+                if (retryInfo?.RetryDelay?.Nanos > 0)
```

Comment:
Why do we not want to use a retry delay of 0? (This may well be moving code I wrote before, but we should check :)